docs/source/llms/llama-index/index.rst

```diff
@@ -0,0 +1,186 @@
+
+.. code-block:: shell
+
+    mkdir -p data
+    curl -L https://raw.githubusercontent.com/run-llama/llama_index/main/docs/docs/examples/data/paul_graham/paul_graham_essay.txt -o ./data/paul_graham_essay.txt
+
+.. code-block:: python
+
+    from llama_index.core import VectorStoreIndex, SimpleDirectoryReader
+
+    documents = SimpleDirectoryReader("data").load_data()
+    index = VectorStoreIndex.from_documents(documents)
+
+Logging the Index to MLflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can log the ``index`` object to the MLflow experiment using the :py:func:`mlflow.llama_index.log_model` function.
+
+One key step here is to specify the ``engine_type`` parameter. The choice of engine type does not affect the index itself,
+but dictates the interface of how you query the index when you load it back for inference.
+
+1. QueryEngine (``engine_type="query"``) is designed for a simple query-response system that takes a single query string and returns a response.
+2. ChatEngine (``engine_type="chat"``) is designed for a conversational agent that keeps track of the conversation history and answer tp the user query based on the context.
+3. Retriever (``engine_type="retriever"``) is a lower-level component that returns the top-k relevant documents matching the query.
+
+
+The following code is an example of logging the index to MLflow with chat engine type.
+
+.. code-block:: python
+
+    import mlflow
+
+    mlflow.set_experiment("llama-index-demo")
+
+    with mlflow.start_run():
+        model_info = mlflow.llama_index.log_model(
+            index,
+            artifact_path="index",
+            engine_type="chat",
+            input_example="What did the author do growing up?",
+        )
+
+.. figure:: ../../_static/images/llms/llama-index/llama-index-artifacts.png
+    :alt: MLflow artifacts for the LlamaIndex index
+    :width: 80%
+    :align: center
+
+.. tip::
+
+    Under the hood, MLflow calls ``as_query_engine()`` / ``as_chat_engine()`` / ``as_retriever()`` method on the index object to convert it to the respective engine instance.
+
+Loading the Index Back for inference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The saved index can be loaded back for inference using the :py:func:`mlflow.pyfunc.load_model` function. This function
+gives an MLflow Python Model backed by the LlamaIndex engine, with the engine type specified during logging.
+
+.. code-block:: python
+
+    import mlflow
+
+    model = mlflow.pyfunc.load_model(model_info.model_uri)
+
+    response = model.predict("What was the first program the author wrote?")
+    print(response)
+    # >> The first program the author wrote was on the IBM 1401 ...
+
+    # The chat engine keeps track of the conversation history
+    response = model.predict("How did the author feel about it?")
+    print(response)
+    # >> The author felt puzzled by the first program ...
+
+
+.. tip::
+
```

Comment:
We might want to pre-emptively add the next logical question to an FAQ section for users. 
"Ok, I saved my LlamaIndex 'model' with a 'chat engine', but now I want to use it as a 'query engine' deployed to an endpoint" - we could walk them through the process of extracting the index and saving with the alternative engine type and showing how the response differs in tracing to fully illustrate the versatility of the design choices for this flavor (the index is huge and expensive to recreate, the engine is easy to swap out - let's just connect the dots for them)

Reply:
Yea I initially added the FAQ section heading and couldn't come up with a good question candidates😅 The engine switching question seems to be a great one to start!

Reply:
Another good one is stuff around settings e.g. how should I change the LLM/embedding model of a loaded index. Not sure how much llama index education we want to do though 🙂